### DISCUSSION TITLE
add custom proxy headers for specific app

### DISCUSSION THREAD

**Discussion by hugokoopmans** [nginx-proxy/nginx-proxy#1632] [Q&A]
Hi,

I am using this compagnion for sometime now, great work!

My issue is the following, I am running a R StudioShiny app behind the nginx proxy, it needs some header settings to keep http socket open but it seems I cannot get it to work.

My main issue is that de docker_gen just overwrites my changes to the nginx.conf.d/default.conf every timeit restarts the container so I cannot make these changes working(or test them)

The stuff I need for one specific server is : 

```
proxy_http_version 1.1;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $connection_upgrade;
proxy_read_timeout 20d;
proxy_buffering off;
client_max_body_size 0;

```

Question : How can I set these parameters in de nginx conf whithout the compagnion overwriting the conf ?

**Comment by buchdag** [comment]
https://github.com/nginx-proxy/nginx-proxy#per-virtual_host
